File: scripts/common_js/config_loader.ts
import { readFileSync, existsSync } from "node:fs";
import { parse } from "smol-toml";

export function getConfigPath(): string {
  const root = process.env.GUOHUB_ROOT;
  if (root) return `${root}/config.toml`;
  return new URL("../../config.toml", import.meta.url).pathname;
}

export function loadConfig(): Record<string, unknown> {
  const path = getConfigPath();
  if (!existsSync(path)) return { defaults: {}, services: {} };
  return parse(readFileSync(path, "utf-8")) as Record<string, unknown>;
}

File: scripts/common_py/config_loader.py
import os
import tomllib
from pathlib import Path

from scripts.common_py.GUOHUB_CLI_CONFIG import NAME_SPACE


def get_config_path() -> Path:
    root = os.environ.get("GUOHUB_ROOT")
    if root:
        return Path(root) / "config.toml"
    return Path(__file__).resolve().parent.parent.parent / "config.toml"


def load_config() -> dict:
    path = get_config_path()
    if not path.exists():
        return {"defaults": {}, "services": {}}
    with open(path, "rb") as f:
        return tomllib.load(f)
